index.bs

```diff
@@ -316,56 +317,94 @@ Interface {#interface}
 =================
 
 <pre class="idl">
-dictionary NavigatorUABrandVersionDict {
-  required DOMString brand;
+dictionary NavigatorUABrandVersion {
+  DOMString brand;
   DOMString version;
 };
 
+dictionary UADataValues {
+  DOMString platform; 
+  DOMString platformVersion;
+  DOMString architecture;
+  DOMString model;
+};
+
 [Exposed=Window]
 interface NavigatorUAData {
-  readonly attribute FrozenArray&lt;NavigatorUABrandVersionDict&gt; brand;
+  readonly attribute FrozenArray&lt;NavigatorUABrandVersion&gt; uaList;
   readonly attribute boolean mobile;
-  Promise&lt;DOMString&gt; getPlatform();
-  Promise&lt;SOMString&gt; getPlatformVersion();
-  Promise&lt;DOMString&gt; getArchitecture();
-  Promise&lt;DOMString&gt; getModel();
+  Promise&lt;UADataValues&gt; getHighEntropyValues(sequence&lt;DOMString&gt; hints);
 };
 
 interface mixin NavigatorUA {
-  [SecureContext] NavigatorUAData getUserAgent();
+  [SecureContext] readonly attribute NavigatorUAData userAgentData;
 };
 Navigator includes NavigatorUA;
 
 </pre>
 
 Processing model {#processing}
 --------------
-<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method MUST run these steps:
+
+<h3 id="monkeypatch-html-windoworworkerglobalscope"><code>WindowOrWorkerGlobalScope</code></h3>
+
+Every {{WindowOrWorkerGlobalScope}} object has an associated <dfn for="WindowOrWorkerGlobalScope">UA list</dfn>, which is a <code><a interface>FrozenArray</a>&lt;<a dictionary>NavigatorUABrandVersion</a>></code>. It is initially the result of [=create a frozen array|creating a frozen array=] from the [=/list=] returned from running [=create UA list=].
+
+<h3 id="create-ua-list-section">Create a UA list</h3>
+
+When asked to run the <dfn>create UA list</dfn> algorithm, the user agent MUST run the following steps:
+1. Let |list| be a [=/list=].
+
+2. Collect pairs of [=user agent/brand=] and [=user agent/significant version=] which represent the user agent,
+    its equivalence class and/or its rendering engine.
+
+3. For each pair:
+
+    1. Let |dict| be a new {{NavigatorUABrandVersion}} dictionary,
+        with [=user agent/brand=] as {{NavigatorUABrandVersion/brand}} and [=user agent/significant version=] as {{NavigatorUABrandVersion/version}}. 
+
+    2. Append |dict| to |list|.
+
+4.  The user agent MAY execute the following steps:
+
+    1.  [=list/Append=] additional items to |list| containing {{NavigatorUABrandVersion}} objects,
+        initialized with arbitrary {{NavigatorUABrandVersion/brand}} and {{NavigatorUABrandVersion/version}} combinations.
+
+    2.  Randomize the order of the items in |list|.
+
+    Note: See [[#grease]] for more details on why these steps might be appropriate.
+
+5. Return |list|.
+
+<h3 id="getters">Getters</h3>
+
+On getting, the {{NavigatorUAData/uaList}} attribute MUST return [=this=]'s [=WindowOrWorkerGlobalScope/UA list=].
```

Comment:
this is a NavigatorUAData, not a WindowOrWorkerGlobalScope. You can either move the UA list to NavigatorUAData, or you can use this's relevant global object's UA list.

Reply:
yeah, missed that. Will change to the latter option